homeassistant/components/sensor/history_average.py

```diff
@@ -0,0 +1,342 @@
+        self._history = []
+        self._state = 0
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Initialize data and register callbacks."""
+        # load data from recorder and do initial calculation
+        yield from self.async_get_history()
+        yield from self.async_update()
+
+        @callback
+        def state_listener(entity, old_state, new_state):
+            """Handle the sensor state changes."""
+            self._history.append(new_state)
+            sorted(self._history, key=attrgetter('last_changed'))
+            self._hass.async_add_job(self.async_update_ha_state(True))
+
+        @callback
+        def sensor_startup(event):
+            """Update on startup."""
+            async_track_state_change(
+                self._hass, self._entity_id, state_listener)
+
+            self._hass.async_add_job(self.async_update_ha_state(True))
+
+        self._hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_START, sensor_startup)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the state is expressed in."""
+        return self._unit_of_measurement
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        # Since the values can/do change with time, not just with
+        # state changes, the sensor should poll
+        return True
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+        def period_in_seconds(period):
+            """Get the period period duration in seconds."""
+            if len(period) != 2 or period[0] == period[1]:
+                return 0.0
+            return (period[1] - period[0]).total_seconds()
+
+        def pretty_duration(period):
+            """Format a duration in days, hours, minutes, seconds."""
+            seconds = period_in_seconds(period)
+            days, seconds = divmod(seconds, 86400)
+            hours, seconds = divmod(seconds, 3600)
+            minutes, seconds = divmod(seconds, 60)
+            if days > 0:
+                return '%dd %dh %dm' % (days, hours, minutes)
+            elif hours > 0:
+                return '%dh %dm' % (hours, minutes)
+            return '%dm' % minutes
+
+        return {
+            ATTR_DURATION: pretty_duration(self._period),
+        }
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return ICON
+
+    @asyncio.coroutine
+    def async_update(self):
+        """Update the sensor's state."""
+        # clean up history & update period
+        yield from self.asysnc_trim_history()
+        start, end = self._period
+
+        _LOGGER.info(" - period start: %s", start)

```

Comment:
This is not `info` log worthy. Max debug. Also limit to 1 statement instead of 2.